### PR TITLE
f-restaurant-card@0.6.0 - Add restaurant cuisines component

### PR DESCRIPTION
**Pull request by 1itvinka**
### Added
- Restaurant cuisines component

Screenshot 2021-10-28 at 17 18 25

Screenshot 2021-10-28 at 17 18 49

Screenshot 2021-10-28 at 17 27 16

## Browsers Tested

- [x] Chrome (latest)
- [ ] Internet Explorer 11
- [ ] Mobile (Please list device/browser – Ideally one iPhone model and one Android)

### CODE REVIEW

**Comment by jamieomaguire** [comment]
I think it would be nice if we created a folder for each subcomponent that could house the component, tests and any extras such as js services.

So:

```
/subcomponents/restaurantcuisines/RestaurantCuisines.vue
/subcomponents/restaurantcuisines/_tests/RestaurantCuisines.test.js 
```

What do you think?

File: packages/components/molecules/f-restaurant-card/src/components/_tests/RestaurantCuisines.test.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        };
+
+        // act
+        const wrapper = shallowMount(RestaurantCuisines, { propsData });
+
+        // assert
+        expect(wrapper.exists()).toBe(true);
+    });
+
+    it('should not be shown if there is no cuisines data', () => {
```
Comment (jamieomaguire) [inline packages/components/molecules/f-restaurant-card/src/components/_tests/RestaurantCuisines.test.js]: this is not a concern for restaurant cuisines, this test should be a part of the restaurantcardv1 suite

File: packages/components/molecules/f-restaurant-card/src/components/_tests/RestaurantCuisines.test.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        };
+
+        // act
+        const wrapper = shallowMount(RestaurantCardV1, { propsData });
+
+        // assert
+        expect(wrapper.find('[data-test-id="restaurant-cuisines"]').exists()).toBe(false);
+    });
+
+    it('should be shown if there is cuisines data', () => {
```
Comment (jamieomaguire) [inline packages/components/molecules/f-restaurant-card/src/components/_tests/RestaurantCuisines.test.js]: same comment as above

File: packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantCuisines.vue [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    </ul>
+</template>
+
+<script>
+export default {
+    name: 'RestaurantCuisines',
+    props: {
+        cuisines: {
+            type: Array,
+            default: null
```
Comment (jamieomaguire) [inline packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantCuisines.vue]: I think that this prob should be required. The consumer should decide whether or not to render cuisines based on their existence in my opinion. Does it make sense to ever render this as an empty component?

File: packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantCuisines.vue [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        }
+    }
+};
+</script>
+
+<style lang="scss" module>
+.c-restaurantCard-cuisines {
+    display: flex;
+    flex-wrap: wrap;
+    margin: 4px 0;
```
Comment (jamieomaguire) [inline packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantCuisines.vue]:
```suggestion
    margin: spacing(x0.5) 0;
```

File: packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantCuisines.vue [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+    & + & {
+        margin-top: 0;
+    }
+}
+
+.c-restaurantCard-cuisines-item {
+    position: relative;
+    padding-right: 10px;
+    margin-right: 5px;
```
Comment (jamieomaguire) [inline packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantCuisines.vue]: Probably best to see if we use spacing() here 
https://github.com/justeat/fozzie/blob/c94a8ee4bcecc8034b4e9d1b9ccdb1057adae91d/src/scss/settings/_variables.scss#L106

File: packages/components/molecules/f-restaurant-card/src/components/_tests/RestaurantCard.v1.test.js [hunk 6]
```diff
@@ -4,7 +4,7 @@ import RestaurantCardV1 from '../RestaurantCard.v1.vue';
 
 describe('RestaurantCard.v1', () => {
     describe('Error Boundary', () => {
-        const slotList = ['cuisines', 'new-label', 'ratings', 'meta-items', 'local-legend', 'badges', 'optional-items'];
+        const slotList = ['new-label', 'ratings', 'meta-items', 'local-legend', 'badges', 'optional-items'];
```
Comment (jamieomaguire) [inline packages/components/molecules/f-restaurant-card/src/components/_tests/RestaurantCard.v1.test.js:7]: we can DRY it up later but for now might be worth adding a test to ensure that an error boundary wraps the new cuisines subcomponent. That is only if we intend to - if we already wrap this in an error boundary on SERP then yeah.

File: packages/components/molecules/f-restaurant-card/stories/RestaurantCard.stories.js [hunk 7]
```diff
@@ -35,7 +30,8 @@ RestaurantCardComponent.args = {
         logoUrl: restaurantLogo,
         imgUrl: restaurantImage,
         isListItem: false,
-        url: 'some-restaurant/12345'
+        url: 'some-restaurant/12345',
+        cuisines: ['Mexican', 'Burgers', 'Chinese']
```
Comment (jamieomaguire) [inline packages/components/molecules/f-restaurant-card/stories/RestaurantCard.stories.js:34]: Cuisines usually have an seo-friendly ID. Does the card component need to know about those? The answer is probably know but just want to be sure!

Comment (1itvinka) [reply packages/components/molecules/f-restaurant-card/stories/RestaurantCard.stories.js:34]: probably not because it's not displayed neither on SERP nor on Home
Screenshot 2021-10-29 at 12 50 19

File: packages/components/molecules/f-restaurant-card/src/components/RestaurantCard.v1.vue [hunk 8]
```diff
@@ -152,6 +157,10 @@ export default {
             type: Boolean,
             default: true
         },
+        cuisines: {
+            type: Array,
+            default: null
```
Comment (minrwhite) [inline packages/components/molecules/f-restaurant-card/src/components/RestaurantCard.v1.vue]: I'd suggest making the default value an empty array and therefore allow simplifying the in-template condition (just use `cuisines.length`) on line 40:
```suggestion
            default: () => []
```

File: packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantCuisines.vue [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+    & + & {
+        margin-top: 0;
+    }
+}
+
+.c-restaurantCard-cuisines-item {
+    position: relative;
+    padding-right: spacing(x1.25);
+    margin-right: spacing(x0.6);
```
Comment (minrwhite) [inline packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantCuisines.vue:42]: I didn't realize we had these spacing multipliers now 🤯

File: packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantCuisines.vue [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+    white-space: nowrap;
+    font-size: $font-body-s-paragraph * 1px;
+
+    &:before {
+        display: none;
+    }
```
Comment (minrwhite) [inline packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantCuisines.vue:48]: Is this needed? i.e is there some pseudo-content that is inserted if you _don't_ add this, or is this here for another reason?

File: packages/components/molecules/f-restaurant-card/stories/RestaurantCard.stories.js [hunk 11]
```diff
@@ -16,11 +16,6 @@ export const RestaurantCardComponent = (args, { argTypes }) => ({
     components: { RestaurantCard },
     props: Object.keys(argTypes),
     template:  `<restaurant-card v-bind="$props">
-                    <template v-slot:cuisines>
-                        <span>pizza</span>
-                        -
-                        <span>burgers</span>
-                    </template>
```
Comment (minrwhite) [inline packages/components/molecules/f-restaurant-card/stories/RestaurantCard.stories.js:23]: ❤️  the use of slots to surface temporary content in stories that will be replaced by subcomponents later; genius!

File: packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantCuisines.vue (hunk 10 above)
Comment (1itvinka) [reply packages/components/molecules/f-restaurant-card/src/components/subcomponents/RestaurantCuisines.vue:48]: for example, on SERP every `li` element have `&:before` with the style. It's for those cases

File: packages/components/molecules/f-restaurant-card/stories/RestaurantCard.stories.js (hunk 11 above)
Comment (1itvinka) [reply packages/components/molecules/f-restaurant-card/stories/RestaurantCard.stories.js:23]: @jamieomaguire :)

File: packages/components/molecules/f-restaurant-card/src/components/RestaurantCard.v1.vue (hunk 8 above)
Comment (jamieomaguire) [reply packages/components/molecules/f-restaurant-card/src/components/RestaurantCard.v1.vue]: Yeah I agree with Ben!

File: packages/components/molecules/f-restaurant-card/stories/RestaurantCard.stories.js (hunk 11 above)
Comment (jamieomaguire) [reply packages/components/molecules/f-restaurant-card/stories/RestaurantCard.stories.js:23]: thank you :D

**Review by minrwhite — APPROVED** [review]

**Review by jamieomaguire — APPROVED** [review]
